internal/providers/azure/azure.go

```diff
@@ -66,29 +64,51 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 // FetchFromOvfDevice has the return signature of platform.NewFetcher. It is
 // wrapped by this and AzureStack packages.
 func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config, report.Report, error) {
-	devicePath := filepath.Join(distro.DiskByIDDir(), configDeviceID)
-
+	var rawConfig []byte
+	var err error
 	logger := f.Logger
-	logger.Debug("waiting for config DVD...")
-	waitForCdrom(logger, devicePath)
 
-	fsType, err := checkOvfFsType(logger, devicePath, ovfFsTypes)
-	if err != nil {
-		return types.Config{}, report.Report{}, err
+	for _, ovfFsType := range ovfFsTypes {
+		device, err := execUtil.GetBlockDevices(ovfFsType)
+		if err != nil {
+			return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve block devices with FSTYPE=%q: %v", ovfFsType, err)
+		}
+		for len(device) > 0 {
+			for i, dev := range device {
+				// verify that this is a CD-ROM device
+				if isCdromPresent(logger, dev) {
+					rawConfig, err = getRawConfig(f, dev, ovfFsType)
+					if len(rawConfig) >= 0 {
+						return util.ParseConfig(logger, rawConfig)
+					} else if err != nil && i == len(device) {
+						logger.Debug("failed to retrieve config: %v", err)
+					}
+				}
+			}
+			// wait for the actual config drive to appear
+			// if it's not shown up yet
+			time.Sleep(time.Second)
+			device, _ = execUtil.GetBlockDevices(ovfFsType)
+		}
 	}
+	return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve config: %v", err)
```

Comment:
We should be retrying indefinitely unless we hit a specific error, so we shouldn't need an error return outside the loop.